### PR TITLE
Fix the formula used for calculating reagents in a plant.

### PR DESCRIPTION
**Pull request by KathyRyals**
## About The Pull Request
There is currently a very crippling bug in botany that makes it so you can't do somewhat complex plants. Why? Because the maximum amout of chemicals in a potency 100 plant is 50u (as intended), and right now, when harvesting a plant with the following traits :

- 50% Chemical A
- 50% Chemical B

You will get a plant with 
- 50u Chemical A
- 0u Chemical B

Here is the deathnettle, that illustrate this issue perfectly.

image

Now for my fixed version.

image

## Why It's Good For The Game

Bug fix good.

## Changelog
:cl: Kathy Ryals, with help from ArcaneMusic
fix: Fixed a bug with the formula used for calculating reagents in a plant. Plants should now contain the correct proportions of chemicals.
/:cl:

### CODE REVIEW

**Comment by Kelenius** [comment]
What. You don't just divide the capacity by 2. Proper fix would be this:

image

**Comment by KathyRyals** [comment]
You mean what ATHATH is proposing? Because I fail to see how that would be better. The division in half is what plants are supposed to get for each reagents, the amount was just doubled up until now.

File: code/modules/hydroponics/seeds.dm [hunk 1]
```diff
@@ -223,7 +223,7 @@
 		CRASH("[T] has no reagents.")
 
 	for(var/rid in reagents_add)
-		var/amount = max(1, round(potency * reagents_add[rid], 1)) //the plant will always have at least 1u of each of the reagents in its reagent production traits
+		var/amount = max(1, round((potency * reagents_add[rid])/2, 1)) //the plant will always have at least 1u of each of the reagents in its reagent production traits
```
Comment (ATH1909) [inline code/modules/hydroponics/seeds.dm:226]: holy shit this is the laziest fix ever

you just effectively cut the amount of chems made by each plant chem trait in half and called it a day lmao

**Comment by ATH1909** [comment]
tldr; this PR is incredibly deceptive and basically just halves the chem production of all plants instead of actually fixing the issue properly

**Comment by ATH1909** [comment]
this doesn't even fix the issue if you crossbreed a bunch of chem production traits onto a plant

File: code/modules/hydroponics/seeds.dm (hunk 1 above)
Comment (MrPerson) [reply code/modules/hydroponics/seeds.dm:226]: Shouldn't this be foo/length(reagents_iadd) instead of foo/2?

Comment (MrPerson) [reply code/modules/hydroponics/seeds.dm:226]: And even that's not right because it assumes each reagent has an equal share.

**Comment by Krysonism** [comment]
A gigantic nerf disguised as a fix.

**Comment by KathyRyals** [comment]
PR lead to a big discussion about this issue on the discord, Arcane is working on a better fix, I'll close this when he uploads his PR.

**Comment by KathyRyals** [comment]
Closed by #51723
